src/renderer/uia/UiaRenderer.cpp

```diff
@@ -104,15 +103,17 @@ CATCH_RETURN();
 // - rectangles - One or more rectangles describing character positions on the grid
 // Return Value:
 // - S_OK
-[[nodiscard]] HRESULT UiaEngine::InvalidateSelection(const std::vector<SMALL_RECT>& rectangles) noexcept
+[[nodiscard]] HRESULT UiaEngine::TriggerSelection(IRenderData* pData) noexcept

```

Comment:
This is a very interesting change, that I'd like to ask @carlos-zamora to verify. 

Ideally I want no code logic change out of this PR. This method, however, does contains logic change, but it's in a *good* way.

Before this PR, this is how `InvalidateSelection` is called in `renderer.cpp`:

```c++
LOG_IF_FAILED(pEngine->InvalidateSelection(_previousSelection));
LOG_IF_FAILED(pEngine->InvalidateSelection(rects));
```

The reason why there's two call to `InvalidateSelection` is that both GdiRendere & DxRenderer needs to first clear the previous selection before draw the current selection. Thais practically not needed inside UiaEngine! 

Also, before this change `_previousSelection` is stored in `renderer.cpp` which UiaEngine has no access to, so it creates another `_prevSelection` internally. This is no longer needed, because `_previousSelection` is now moved inside `RenderEngineBase`.

I see this as another example why a thinner `renderer.cpp` is beneficial, because it allows more control in the engine side. And different engines do have different needs when responding to the same event.